pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -483,14 +483,15 @@ func (m *kubeGenericRuntimeManager) executePreStopHook(pod *v1.Pod, containerID
 	glog.V(3).Infof("Running preStop hook for container %q", containerID.String())
 
 	start := metav1.Now()
-	done := make(chan struct{})
+	done := make(chan bool)
 	go func() {
 		defer close(done)
```

Comment:
What is it diff from below `done <- true` notification ?

Reply:
I'm not sure what you mean? The diff? I needed to send some kind of notification to flag that preStop hook is done and struct{} seemed less than ideal for this job?

Reply:
I only see you send true. Close channel has the same effect.

Reply:
Uh, this might actually be my Go knowledge working against me. 
Thanks for enlighten me. :)

Reply:
I'll put in more work to reproduce this with a test...